### PR TITLE
Add message broker (RabbitMQ)

### PR DESCRIPTION
**Pull request by Rediska235**

### CODE REVIEW

File: MusicApp.Identity.Infrastructure/Repositories/RoleRepository.cs [hunk 1]
```diff
@@ -7,15 +7,15 @@ namespace MusicApp.Identity.Infrastructure.Repositories;
...
+    private readonly AppDbContext _appContext;
 
-    public RoleRepository(AppDbContext db)
+    public RoleRepository(AppDbContext appContext)
     {
-        _db = db;
+        _appContext = appContext;
     }
 
     public async Task<Role> GetRoleByTitleAsync(string title)
```
Comment (gfady) [inline MusicApp.Identity.Infrastructure/Repositories/RoleRepository.cs]: Task<Role?>

File: MusicApp.Identity.Infrastructure/Repositories/UserRepository.cs [hunk 2]
```diff
@@ -7,30 +7,30 @@ namespace MusicApp.Identity.Infrastructure.Repositories;
...
+    private readonly AppDbContext _appContext;
 
-    public UserRepository(AppDbContext db)
+    public UserRepository(AppDbContext appContext)
     {
-        _db = db;
+        _appContext = appContext;
     }
 
     public async Task<User> GetUserByUsernameAsync(string username)
```
Comment (gfady) [inline MusicApp.Identity.Infrastructure/Repositories/UserRepository.cs]: Task<User?> and check it through all files

File: MusicApp.Identity.Web/Filters/LoginValidationFilterAttribute.cs [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+
+    public LoginValidationFilterAttribute()
+    {
+        _validator = new();
+    }
+    
+    public override void OnActionExecuting(ActionExecutingContext actionContext)
+    {
+        object playlistObj;
+        actionContext.ActionArguments.TryGetValue("userLoginDto", out playlistObj);
```
Comment (gfady) [inline MusicApp.Identity.Web/Filters/LoginValidationFilterAttribute.cs]: you don't have to declare playlistObj and cast it to UserRegisterDto outside the TryGetValue method. and it's better to check if playlistObj is not null here

File: MusicApp.Identity.Web/Filters/RegisterValidationFilterAttribute.cs [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+
+    public RegisterValidationFilterAttribute()
+    {
+        _validator = new();
+    }
+    
+    public override void OnActionExecuting(ActionExecutingContext actionContext)
+    {
+        object playlistObj;
+        actionContext.ActionArguments.TryGetValue("userRegisterDto", out playlistObj);
```
Comment (gfady) [inline MusicApp.Identity.Web/Filters/RegisterValidationFilterAttribute.cs]: the same

File: MusicApp.PlaylistService.Web/MessageBroker/SongConsumer.cs [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+
+            case Operation.Updated:
+                await _service.UpdateSongAsync(context.Message.Song, new CancellationToken());
+                break;
+
+            case Operation.Deleted:
+                await _service.RemoveSongAsync(context.Message.Song, new CancellationToken());
+                break;
+        }
+
```
Comment (gfady) [inline MusicApp.PlaylistService.Web/MessageBroker/SongConsumer.cs]: extra line

Comment (Rediska235) [reply MusicApp.PlaylistService.Web/MessageBroker/SongConsumer.cs]: I removed everything related to song messages. And now playlist service can get songs via grpc.

**Review by gfady — APPROVED** [review]

**Review by KatsiarynaAnashkevich — APPROVED** [review]

**Pull request merged**